### PR TITLE
Add error handling for sentry events

### PR DESCRIPTION
**Pull request by anurags92**
Add error handling for Sentry [event](https://sentry.io/organizations/dgraph/issues/1577515284/events/1ce909f7c45a4e52930c235a033d5908/?project=1805390). Also add more logs to help in reproducing error and putting defensive checks in place.

This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1322)

### CODE REVIEW

File: value.go [hunk 1]
```diff
@@ -1515,6 +1515,11 @@ func (vlog *valueLog) Read(vp valuePointer, s *y.Slice) ([]byte, func(), error)
 			return nil, cb, err
 		}
 	}
+	if uint32(len(kv)) <= h.klen+h.vlen {
```
Comment (gja) [inline value.go:1518]: <, not <=

File: iterator.go [hunk 2]
```diff
@@ -169,6 +169,8 @@ func (item *Item) yieldItemValue() ([]byte, func(), error) {
 		vp.Decode(item.vptr)
 		result, cb, err := item.db.vlog.Read(vp, item.slice)
 		if err != ErrRetry {
+			//item.db.opt.Logger.Errorf("Unable to read: Key: %v, Version : %v, meta: %v userMeta: %v",
```
Comment (parasssh) [inline iterator.go:172]: remove dead code.

Comment (anurags92) [reply iterator.go:172]: Thanks, but I was testing this on TeamCity. This needs to go in.

**Comment by anurags92** [comment]
> > _[Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1322)_ status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @anurags92, @ashish-goswami, @manishrjain, and @parasssh)
> 
> _[iterator.go, line 172 at r4](https://reviewable.io/reviews/dgraph-io/badger/1322#-M6ca0BY7bxbhQlnZLHP:-M6ca0BY7bxbhQlnZLHQ:br9pwom) ([raw file](https://github.com/dgraph-io/badger/blob/0444fd80c5d4e62f1a89b1a3c5f1e35844b60cd4/iterator.go#L172)):_
> 
> > ```go
> > 		result, cb, err := item.db.vlog.Read(vp, item.slice)
> > 		if err != ErrRetry {
> > 			if result == nil && cb == nil {
> > ```
> 
> We want to see this log message when there's an error. `if err != nil` would be a better way of checking it.

Done, thanks.

**Pull request merged**